File: server/src/main/java/net/progruzovik/dissent/captain/Captain.java
package net.progruzovik.dissent.captain;

import net.progruzovik.dissent.battle.model.Battle;
import net.progruzovik.dissent.battle.model.Side;
import net.progruzovik.dissent.model.entity.Ship;
import net.progruzovik.dissent.model.event.Event;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;

import java.util.List;
import java.util.function.Consumer;

public interface Captain extends Consumer<Event<?>> {

    @NonNull
    String getId();

    @NonNull
    List<Ship> getShips();

    @Nullable
    Battle getBattle();

    void addToBattle(@NonNull Side side, @NonNull Battle battle);
}

File: server/src/main/kotlin/net/progruzovik/dissent/captain/Captain.kt
package net.progruzovik.dissent.captain

import net.progruzovik.dissent.model.domain.Ship
import net.progruzovik.dissent.model.domain.battle.Battle
import net.progruzovik.dissent.model.domain.battle.Side

interface Captain {

    val id: String

    val ships: List<Ship>

    val battle: Battle?

    fun addToBattle(battle: Battle, side: Side)

    fun endTurn()
}
